ui/static/ui/js/manage_taxonomies.jsx

```diff
@@ -338,11 +431,28 @@ define('manage_taxonomies', ['react', 'lodash', 'jquery', 'utils'],
       });
       this.setState({vocabularies: vocabularies});
     },
+    updateTerm: function(vocabSlug, term) {
+      var vocabularies = this.state.vocabularies;
+      var vocabIndex = _.findIndex(vocabularies, function(vocabularyObj) {
+        return vocabularyObj.vocabulary.slug === vocabSlug;
+      });
+      if (vocabIndex > -1) {
+        var terms = vocabularies[vocabIndex].terms;
+        var termIndex = _.findIndex(terms, function(termObj) {
+          return termObj.id === term.id;
+        });
+        if (termIndex > -1) {
+          vocabularies[vocabIndex].terms[termIndex] = term;
```

Comment:
You can't ever modify `this.state` or anything attached to it, you have to clone the object first. Why not do something like:

      vocabularies = _.map(vocabularies, function(tuple) {
        if (tuple.vocabulary.slug === vocabSlug) {
          var terms = _.map(tuple.terms, function(tupleTerm) {
            if (tupleTerm.id === term.id) {
              return term;
            }
            return tupleTerm;
          });
          return {
            terms: terms,
            vocabulary: tuple.vocabulary
          };
        }